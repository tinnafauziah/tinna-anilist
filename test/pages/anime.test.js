import flushPromises from 'flush-promises';
import animeIndex from '~/pages/anime';
import animeMock from '~/test/mocks/store-mocks/anime-mocks';

let store;
const mocks = {
  $router: {
    push: jest.fn(),
  },
  $route: {
    params: {
      id: 123
    }
  }
};
const modules = { ...animeMock };

describe('Anime Index Page', () => {
  const mockPage = () => {
    store = global.createStore({ ...modules });
    return global.shallowMount(animeIndex, { store, mocks });
  };
  it('should anime Index page match snapshot', async () => {
    const wrapper = mockPage();
    await flushPromises();
    expect(wrapper.element).toMatchSnapshot();
  });
});
