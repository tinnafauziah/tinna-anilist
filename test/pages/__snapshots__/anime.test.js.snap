// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anime Index Page should anime Index page match snapshot 1`] = `
<div
  class="anime"
>
  <nuxtchild-stub />
</div>
`;
